salt/states/locale.py

```diff
@@ -37,21 +42,26 @@ def system(name):
            'changes': {},
            'result': None,
            'comment': ''}
-    if __salt__['locale.get_locale']() == name:
-        ret['result'] = True
-        ret['comment'] = 'System locale {0} already set'.format(name)
-        return ret
-    if __opts__['test']:
-        ret['comment'] = 'System locale {0} needs to be set'.format(name)
-        return ret
-    if __salt__['locale.set_locale'](name):
-        ret['changes'] = {'locale': name}
-        ret['result'] = True
-        ret['comment'] = 'Set system locale {0}'.format(name)
-        return ret
-    else:
+    try:
+        if __salt__['locale.get_locale']() == name:
+            ret['result'] = True
+            ret['comment'] = 'System locale {0} already set'.format(name)
+            return ret
+        if __opts__['test']:
+            ret['comment'] = 'System locale {0} needs to be set'.format(name)
+            return ret
+        if __salt__['locale.set_locale'](name):
+            ret['changes'] = {'locale': name}
+            ret['result'] = True
+            ret['comment'] = 'Set system locale {0}'.format(name)
+            return ret
+        else:
+            ret['result'] = False
+            ret['comment'] = 'Failed to set system locale to {0}'.format(name)
+            return ret
+    except CommandNotFoundError as err:
```

Comment:
Ah I copied it from a different one to avoid making a mistake in the except -_- I copied the wrong one. I will have a follow up PR in a few minutes once I am awake.